packages/js/src/ui/components/Inbox.tsx

```diff
@@ -10,9 +11,44 @@ type InboxProps = {
   renderBell?: ({ unreadCount }: { unreadCount: number }) => JSX.Element;
 };
 
-export const Inbox = (props: InboxProps) => {
+const InboxContent = () => {
   const [currentScreen, setCurrentScreen] = createSignal<'inbox' | 'settings'>('inbox');
   const { t } = useLocalization();
+
+  const novu = useNovu();
+
+  const [pages, initialLoading, setEl, { end }] = createInfiniteScroll(async (page) => {
+    return (await novu.feeds.fetch({ page })).data;
+  });
```

Comment:
☑ **chore:** this hook and some other conditional rendering/loading logic should be a part of the `Notifications` component.